src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -57,20 +56,35 @@ public BtConnectorHelper() {
         this.myPeerName = peerName;
         this.lastAvailableList.clear();
         Stop();
-        mBTConnector = new BTConnector(context,this,this,conSettings,instanceEncryptionPWD);
+        mBTConnector = new BTConnector(context,this,this,conSettings);
         return mBTConnector.Start(this.myPeerIdentifier,this.myPeerName);
     }
 
+    public boolean isRunning(){
+        if(mBTConnector != null){
+            return true;
+        }else{
+            return false;
+        }
+    }
     public void Stop(){
         if(mBTConnector != null){
             mBTConnector.Stop();
             mBTConnector = null;
         }
 
-        if (mBtToServerSocket != null) {
-            print_debug("Disconnect:::Stop : mBtToServerSocket");
-            mBtToServerSocket.Stop();
-            mBtToServerSocket = null;
+        if (mServerSocketList != null) {
+            for(int i = 0; i < mServerSocketList.size(); i++) {
```

Comment:
List isn't safe for concurrent access so if two threads both call stop then bad things are going to happen.